staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -770,6 +778,49 @@ func (vs *VSphere) InstanceShutdownByProviderID(ctx context.Context, providerID
 	return !isActive, nil
 }
 
+// InstanceMetadataByProviderID returns metadata of the specified instance.
+func (vs *VSphere) InstanceMetadataByProviderID(ctx context.Context, providerID string) (*cloudprovider.InstanceMetadata, error) {
+	if providerID == "" {
+		return nil, fmt.Errorf("couldn't compute InstanceMetadata for empty providerID")
+	}
+
+	// TODO not get nodeName by GetNodeNameFromProviderID here. If it not behave expected,
+	// get nodeName by vm.GetNodeNameFromProviderID.
+	nodeName := providerID
+	md, err := vs.instanceMetadataByNodeName(ctx, nodeName)
+	if vclib.IsManagedObjectNotFoundError(err) {
```

Comment:
It wasn't there in NodeAddresses nor NodeAddressesByProviderID.

let's remove (at least for now) - to make that a no-op change.

Reply:
done.